library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -531,12 +531,13 @@ impl<T> VecDeque<T> {
     ///
     /// let deque: VecDeque<u32> = VecDeque::new();
     /// ```
-    // FIXME: This should probably be const
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
+    #[rustc_const_stable(feature = "const_vec_deque_new", since = "CURRENT_RUSTC_VERSION")]
     #[must_use]
-    pub fn new() -> VecDeque<T> {
-        VecDeque::new_in(Global)
+    pub const fn new() -> VecDeque<T> {
+        // FIXME: This should just be `VecDeque::new_in(Global)` once that's stable.
```

Comment:
You could also slap `#[allow_internal_unstable(allocator_api)]` onto `VecDeque::new` instead. Not sure if that's frowned upon though. What does the library team say?